### PR TITLE
Refactor input components to use styled-system

### PR DESCRIPTION
**Pull request by sapegin**
- [x] Input
  - [x] Remove fullWith prop, make full width by default
- [x] Select
  - [x] Remove fullWith prop, make full width by default
  - [x] Use Icon component instead of backgfound (#46)
- [x] Button (#56)
  - [x] Remove fullWith prop, keep inline-block by default
- [x] Remove usages of fullWidth prop

### CODE REVIEW

File: src/components/primitives/Select/Select.js [hunk 1]
```diff
@@ -1,48 +1,64 @@
+import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
-import { getDownArrowIcon } from './icons';
+import { css } from '@styled-system/css';
+import Flex from '../Flex';
+import IconBase from '../Icon';
+
+const Icon = styled(IconBase)({
```
Comment (okonet) [inline src/components/primitives/Select/Select.js:8]: I would instead use `css` prop in the render methods on the component

Comment (sapegin) [reply src/components/primitives/Select/Select.js:8]: Looks like it doesn't work in Styleguidist or there's a problem somewhere else when I import from `*/macros`. I had the same error about `fs` that we've seen on the workshop ;-/

**Comment by sapegin** [comment]
I think I'm done here 🦊

File: src/components/primitives/Button/Button.js [hunk 2]
```diff
@@ -1,59 +1,66 @@
...
-	font-size: ${props => props.theme.fontSizes.md};
-	color: ${props => props.theme.colors[getColor(props.variant)]};
-	background-color: ${props =>
-		props.theme.colors[getBgColor(props.variant)] || 'transparent'};
-	text-decoration: none;
-	user-select: none;
-	box-sizing: border-box;
+const Button = styled.button(
+	css({
+		boxBizing: 'border-box',
```
Comment (okonet) [inline src/components/primitives/Button/Button.js]: Typo: `boxBizing` → `boxSizing`

File: src/components/primitives/Button/Button.js [hunk 3]
```diff
@@ -1,59 +1,66 @@
...
+		display: 'inline-block',
+		height: '2.5rem',
```
Comment (okonet) [inline src/components/primitives/Button/Button.js]: Does Button needs height set? What is going to happen if the label is multi-line?

File: src/components/primitives/Card/Card.md [hunk 4]
```diff
@@ -4,6 +4,7 @@ By using subcomponents `Card.Cover`, `Card.Body` and `Card.Footer` users can cre
 
 ```jsx harmony
 import Box from '../Box';
+import Flex from '../Flex';
```
Comment (okonet) [inline src/components/primitives/Card/Card.md]: Why is this import needed? It doesn't look there are any usages added.

File: src/components/primitives/Select/Select.js (hunk 1 above)
Comment (okonet) [reply src/components/primitives/Select/Select.js:8]: Can you file an issue about it so I can take a look?

**Review by okonet — CHANGES_REQUESTED** [review]

File: src/components/primitives/Button/Button.js (hunk 3 above)
Comment (sapegin) [reply src/components/primitives/Button/Button.js]: To make all form elements have the same height in all browsers.

Comment (sapegin) [reply src/components/primitives/Button/Button.js]: I guess it could be `min-height` instead.

Comment (okonet) [reply src/components/primitives/Button/Button.js]: Yes, `min-height` sounds more appropriate

File: src/components/primitives/Select/Select.js (hunk 1 above)
Comment (sapegin) [reply src/components/primitives/Select/Select.js:8]: https://github.com/component-driven/component-driven-development/issues/60

**Comment by sapegin** [comment]
Done with the fixes 🦆

**Review by okonet — APPROVED** [review]

**Review by okonet — APPROVED** [review]
